calc.js

```diff
@@ -4,150 +4,217 @@ const keys = document.querySelector('#keyboard');
+  { order: 2, value: '*', classes: ARITHMETIC_KEYBOARD_CLASS, dataset: true },
+  { order: 3, value: '-', classes: ARITHMETIC_KEYBOARD_CLASS, dataset: true },
+  { order: 4, value: '+', classes: ARITHMETIC_KEYBOARD_CLASS, dataset: true },
+  { order: 5, value: '=', classes: ARITHMETIC_KEYBOARD_CLASS, dataset: true },
+]
+
+const FUNCTIONS_KEYBOARD = [
+  { order: 0, value: OPERATIONS.PERCENT, classes: FUNCTION_KEYBOARD_CLASS },
+  { order: 1, value: OPERATIONS.CE, classes: FUNCTION_KEYBOARD_CLASS },
+  { order: 2, value: OPERATIONS.C, classes: FUNCTION_KEYBOARD_CLASS },
+  { order: 3, value: OPERATIONS.DEVIDE_BY_X, classes: FUNCTION_KEYBOARD_CLASS },
+  { order: 4, value: 'x<sup>2</sup>', classes: FUNCTION_KEYBOARD_CLASS, html: true },
+  { order: 5, value: '&#8730;', classes: SQRT_KAYBOARD_CLASS, html: true },
+]
+
+const renderButtons = (root, elements) => {
+  elements.sort((a, b) => a.order - b.order).forEach(({ value, classes, dataset, html }) => {
+    const div = document.createElement('div')
+    div.attributes;
+    if (html) {
+      div.innerHTML = value
+    } else {
+      div.textContent = value;
+    }
 
+    if (dataset) div.dataset.operator = value;
+    div.classList.add(...classes)
+    root.appendChild(div)
+  });
+
+}
+
+const initialize = () => {
+  renderButtons(keys, CALCULATOR_KEYBOARD)
+  renderButtons(arithmeticKeys, ARITHMETIC_KEYBOARD)
+  renderButtons(functionKeys, FUNCTIONS_KEYBOARD)
+}
 
 // ----=== FUNCTIONS ===----- 
+const functionKeysOperations = (operation) => {
+  switch (operation) {
 
-function getPress(event) {
+    case OPERATIONS.PERCENT:
+      return display(OPERATIONS.PERCENT);   // not ready
 
-  if ( event.target.classList.contains('sqrt')  ) {  // functions listener
-
-    return display( Math.sqrt(obiekt.value));
-
-  } else if ( event.target.classList.contains('funct') ) {   // functions listener
-  
-    switch(event.target.textContent) {
-      
-      case '%':
-        return display ( '%' );   // not ready
-        
-      case 'CE':
-        return display ( 'CE');   // not ready
-
-      case 'C':
-        clear();
-        return display(0);
-      
-      case '1/x':
-        return display( 1/obiekt.value );   
-
-      case 'x2':
-        return display( exponent( obiekt.value ) );
-    }
+    case OPERATIONS.CE:
+      return display(OPERATIONS.CE);   // not ready
 
-  } // end of functions listener 
+    case OPERATIONS.C:
+      clear();
+      return display(0);
 
+    case OPERATIONS.DEVIDE_BY_X:
+      return display(1 / obiekt.value);
 
-const dataOp = document.querySelectorAll('[data-operator]');
-  
-for ( let i = 0; i < dataOp.length ; i++) {
-  if ( event.target.textContent === dataOp[i].textContent ) {
-    obiekt.operator = dataOp[i].textContent;
-    return operatorCheck(event.target.textContent);
+    case 'x2':
+      return display(exponent(obiekt.value));
   }
-
 }
+function getPress(event) {
```

Comment:
Jak przekształcam tą funkcję na wzór functionKeysOperation to mi sie rozsypuje kalkulator i nie ma przycisków i nie mogę pojąć dlaczego.. Nie wiem jak to ugryźć